node-gateway.js

```diff
@@ -111,30 +111,39 @@ class NodeGateway extends ApolloGateway {
     // This is a local schema, but we treat it as if it were a remote schema,
     // because all other schemas are (probably) remote. In that case, we need
     // to provide the Federated SDL as part of the type definitions.
-    const res = graphqlSync({
-      schema: nodeSchema,
+    const typeDefs = parse(graphqlSync({
+      schema: this.nodeSchema,
       source: 'query { _service { sdl } }',
-    });
+    }).data._service.sdl);
 
-    defs.serviceDefinitions.push({
-      typeDefs: parse(res.data._service.sdl),
-      schema: nodeSchema,
+    return super.createSupergraphFromSubgraphList([...subgraphs, {
       name: NODE_SERVICE_NAME,
-    });
+      typeDefs,
+    }]);
+  }
 
-    return defs;
+  createNodeDataSource() {
+    return new LocalGraphQLDataSource(this.nodeSchema);
   }
+}
 
+/**
+ * An ApolloGateway which provides `Node` resolution across all federated
+ * services, and a global `node` field, like Relay.
+ */
+class NodeGateway extends ApolloGateway {
   /**
    * Override `createDataSource` to let the local Node resolution service be
    * created without complaining about missing a URL.
    */
   createDataSource(serviceDef) {
-    if (serviceDef.schema) {
-      return new LocalGraphQLDataSource(serviceDef.schema);
+    const { supergraphSdl } = this.config
+    if (serviceDef.name === NODE_SERVICE_NAME && supergraphSdl instanceof NodeCompose) {
```

Comment:
`serviceDef` only contains `name` and `url` fields. So, we can't access `serviceDef.schema` anymore